### DISCUSSION TITLE
A little confused on what to host on TFTP server

### DISCUSSION THREAD

**Discussion by icsy7867** [ipxe/ipxe#949] [General]
Sorry for this newbie post.  I recently have compiled/configured an ISO which is working beautifully.

```
make bin/ipxe.lkrn bin-x86_64-efi/ipxe.efi EMBED=boot.ipxe TRUST=incommon.ca.crt
./util/genfsimg -o ipxe-menu.iso bin/ipxe.lkrn bin-x86_64-efi/ipxe.efi
```

This boots and does exactly what I want it to.  So now I am trying to figure out what I need to host these files via TFTP so DHCP can use them to push out to our network segments.  I am a little confused on which files I actually need.  Would I just need the .efi file to be hosted?  Or is there any sort of guidance on what to generate using the ipxe tools?

**Comment by NiKiZe** [comment]
Most BIOS or firmware implementations only supports downloading files over TFTP, so what you need to have on your TFTP is simply the files that you tell DHCP to hand out.
For EFI this is ipxe.efi (please take note that it different binaries between 64bit intel, and 32bit intel, there is also other platforms)
For any legacy pcbios you will probably want undionly.kpxe.

Depending on your network and your devices on the network you might need different things.
Start with one thing, such as EFI boot using ipxe.efi, and then add from there.

Once iPXE is running then http is usually supported, so any scripts or other files is often better served over http instead of TFTP, but it is still possible to keep these thing on TFTP as well.

**Reply by icsy7867** [reply]
Great, so I basically would just need to server the .efi file?

Currently I have a .lkrn file.  Would this not be supported over the .efi file?

**Reply by NiKiZe** [reply]
the .lkrn file can be booted by syslinux or pxelinux, not by a bios or undi stack directly, for that you need the .kpxe or .pxe targets.
the valid builds for those are undionly.kpxe and ipxe.pxe. With some risk of making things more confusing the fuller list is available at https://ipxe.org/appnote/buildtargets - since you dont need these at this time, don't care about them until you do.
